decentraland/unity-explorer
# Feat: screencapture HUD for camera reel

## What does this PR change?

It brings Screencapture HUD with respective buttons and visual effects. Also several polishing things were maid. 
It includes:
- sidebar button for entering camera mode
- hiding all UI when entering camera mode 
- blocking and resetting raycasts when in camera mode
- screencapture VFX when capturing screenshot
- fixed cursor lock/unlock that is similar to normal cursor behaviour

## How to test the changes?
1. Launch the explorer and explore screencapture camera mode
2. Open it via sidebar button and verify it behaves correctly
3. Open-close via C button
4. Exploratory testing of opening/closing via different modes and opening other tabs
5. Capture screenshot via button and via shortcut - verify VFX and splash going down. 
6. Shortcuts menu open/close

Verify that without proper args it is not presented (sidebar and shortcuts doesn't work)

Note: known issues for further polishing
- not all shortcuts are blocked when in camera mode
- audio effects are behaving not always properly
- vfx overlaps camera reel button at the end

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
